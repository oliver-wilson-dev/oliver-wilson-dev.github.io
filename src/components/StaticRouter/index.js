import React from 'react';
import PropTypes from 'prop-types';
import { StaticRouter as ReactDomStaticRouter } from 'react-router-dom';

const StaticRouter = ({ children, location, context }) => (
  <ReactDomStaticRouter context={context} location={location}>
    {children}
  </ReactDomStaticRouter>
);

StaticRouter.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  location: PropTypes.string,
  context: PropTypes.shape(),
};

export default StaticRouter;
