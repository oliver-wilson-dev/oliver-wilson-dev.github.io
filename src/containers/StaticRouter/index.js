import { connect } from 'react-redux';
import StaticRouter from '../../components/StaticRouter';
import { getLocation } from '../../state/selectors';

const mapStateToProps = state => ({
  location: getLocation(state)
});

export default connect(mapStateToProps)(StaticRouter);
