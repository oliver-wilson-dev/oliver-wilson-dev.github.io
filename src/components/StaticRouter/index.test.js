import React from 'react';
import { shallow } from 'enzyme';
import StaticRouter from './index';

jest.mock('react-router-dom', () => ({
  StaticRouter: ({ children }) => <div>{children}</div>,
}));

const defaultProps = {
  location: 'test-location',
  context: {}
};

const render = () => shallow(
  <StaticRouter {...defaultProps}>
    <h1>hello world</h1>
  </StaticRouter>
);

describe('StaticRouter', () => {
  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });
});
