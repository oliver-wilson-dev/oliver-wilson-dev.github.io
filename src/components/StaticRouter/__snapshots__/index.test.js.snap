// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticRouter should render correctly 1`] = `
<StaticRouter
  context={Object {}}
  location="test-location"
>
  <h1>
    hello world
  </h1>
</StaticRouter>
`;
